### PR TITLE
Купцова -- Лабораторная работа #0

### PR DESCRIPTION
**Pull request by alesya-kuptsova**

### CODE REVIEW

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        String numstr = valueOf(num);
+        assertEquals(im, numstr);
+    }
+
+    @Test
+    public void canMultiplyNumbersWithNegativePartS() {
+        ComplexNumber z1 = new ComplexNumber(-1, 4);
+        ComplexNumber z2 = new ComplexNumber(-2, -3);
+        ComplexNumber num = z1.multiply(z2);
+        assertEquals(new ComplexNumber(14, -5), num);
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: You need to take _new ComplexNumber(14, -5)_ to somethink like "expectedValue" variable,
and _num_ to "actualValue" variable everywhere

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        String numstr = valueOf(num);
+        assertEquals(re, numstr);
+    }
+    @Test
+    public void canGetImaginaryFromComplexNumberWithImaginaryPart() {
+        String re = "3.14";
+        String im = "-13.0";
+        ComplexNumber z = new ComplexNumber(re, im);
+        double num = z.getImaginary();
+        String numstr = valueOf(num);
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: please use the lowerCamelCase everywhere

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Test
+    public void canNotCreateComplexNumberWithUncorrectValues() {
+        String re = "3.14";
+        String im = "-rjfho;i";
+        thrown.expect(IllegalArgumentException.class);
+        ComplexNumber z = new ComplexNumber(re, im);
+    }
+    @Test
+    public void canGetRealFromComplexNumberWithARealPart() {
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: change name please, what's the 'A' in ...WithARealPart ?

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        String re = "3.14";
+        String im = "-13.0";
+        ComplexNumber z = new ComplexNumber(re, im);
+        double num = z.getReal();
+        String numstr = valueOf(num);
+        assertEquals(re, numstr);
+    }
+    @Test
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java:26]: add blank line after each test

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        thrown.expect(IllegalArgumentException.class);
+        ComplexNumber z = new ComplexNumber(re, im);
+    }
+    @Test
+    public void canGetRealFromComplexNumberWithARealPart() {
+        String re = "3.14";
+        String im = "-13.0";
+        ComplexNumber z = new ComplexNumber(re, im);
+        double num = z.getReal();
+        String numstr = valueOf(num);
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: it is not necessary to convert numbers into strings

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        String re = "3.14";
+        String im = "-13.0";
+        ComplexNumber z = new ComplexNumber(re, im);
+        double num = z.getImaginary();
+        String numstr = valueOf(num);
+        assertEquals(im, numstr);
+    }
+
+    @Test
+    public void canMultiplyNumbersWithNegativePartS() {
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: remove 'S' at the end of test name and write 's'. Please

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        ComplexNumber z1 = new ComplexNumber(-1, 4);
+        ComplexNumber z2 = new ComplexNumber(-2, -3);
+        ComplexNumber num = z1.multiply(z2);
+        assertEquals(new ComplexNumber(14, -5), num);
+    }
+}
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java:43]: add blank line at the end of file

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import ru.unn.agile.ComplexNumber.model.ComplexNumber;
+
+import static java.lang.String.valueOf;
+import static junit.framework.TestCase.assertEquals;
+
+public class KuptsovaAlesyaTest {
+    @Rule
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: i think you need to add blank line before _@Rule_

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void canNotCreateComplexNumberWithUncorrectValues() {
+        String re = "3.14";
+        String im = "-rjfho;i";
+        thrown.expect(IllegalArgumentException.class);
```
Comment (YuferevKirill) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/KuptsovaAlesyaTest.java]: can you rewrite test to somethink like that:
    @Test(expected = IllegalArgumentException.class)
    public void canNotCreateComplexNumberWithUncorrectValues() {
          your_test
    }

**Review by YuferevKirill — CHANGES_REQUESTED** [review]

**Comment by alesya-kuptsova** [comment]
@mkildishev

**Comment by alesya-kuptsova** [comment]
@YuferevKirill , fixed! Thanks for your comments!)
@mkildishev

**Comment by mkildishev** [comment]
👍

**Comment by vgvoleg** [comment]
👍

**Pull request merged**